appveyor.yml

```diff
@@ -0,0 +1,170 @@
+      $env:APPVEYOR_PULL_REQUEST_NUMBER -and \
+      $env:APPVEYOR_BUILD_NUMBER -ne (\
+        (\
+          Invoke-RestMethod \
+            ` https://ci.appveyor.com/api/projects/\
+                $env:APPVEYOR_ACCOUNT_NAME/\
+                $env:APPVEYOR_PROJECT_SLUG/\
+                history?recordsNumber=50\
+        ).builds | ` Where-Object pullRequestId -eq \
+                                  $env:APPVEYOR_PULL_REQUEST_NUMBER\
+      )[0].buildNumber\
+    ) { ` \
+      throw \"There are newer queued builds \
+              for this pull request, failing early.\" \
+    }"
+  - cmd: ECHO "Filesystem root:"
+  - ps: "ls \"C:/\""
+  - cmd: ECHO "Installed SDKs:"
+  - ps: "ls \"C:/Program Files/Microsoft SDKs/Windows\""
+
+  # Prepend newly installed Python to the PATH of this build (this cannot be
+  # done from inside the powershell script as it would require to restart
+  # the parent CMD process).
+  - cmd: "SET PATH=%PYTHON%;%PYTHON%\\Scripts;%PATH%"
+
+  # Check that we have the expected version and architecture for Python
+  - cmd: "python --version"
+  - cmd: "\
+      python -c \"import struct; \
+                  print('    {}-bit'.format(struct.calcsize('P') * 8))\""
+
+  # Upgrade to the latest version of pip to avoid it displaying warnings
+  # about it being out of date.
+  - cmd: "pip install --disable-pip-version-check --user --upgrade pip"
+  - cmd: "pip install wheel"
+  - cmd: "cinst cmake"
+
+  # Install the build dependencies of the project. If some dependencies contain
+  # compiled extensions and are not provided as pre-built wheel packages,
+  # pip will build them from source using the MSVC compiler matching the
+  # target Python version and architecture
+  - cmd: "%CMD_IN_ENV% pip install -r requirements.txt"
+  - cmd: "%CMD_IN_ENV% pip install -r requirements-dev.txt"
+
+build_script:
+  # Build the compiled extension
+  - cmd: "%CMD_IN_ENV% python setup.py build"
+
+test_script:
+  # workaround for CMake not wanting sh.exe on PATH for MinGW
+  - cmd: "set PATH=%PATH:C:\\Program Files\\Git\\usr\\bin;=%"
+  - cmd: "set PATH=%PATH:C:\\Program Files (x86)\\Git\\usr\\bin;=%"
+  - cmd: "set PATH=%PATH:C:\\Program Files\\Git\\bin;=%"
+  - cmd: "set PATH=%PATH:C:\\Program Files (x86)\\Git\\bin;=%"
+  - cmd: "set PATH=%PATH:C:\\Program Files\\Git\\cmd;=%"
+  - cmd: "set PATH=%PATH:C:\\Program Files (x86)\\Git\\cmd;=%"
+  - cmd: "%CMD_IN_ENV% python -m nose -v -w tests %EXTRA_TEST_ARGS%"
+
+after_test:
+  # If tests are successful, create binary packages for the project.
+  - cmd: "%CMD_IN_ENV% python setup.py bdist_wheel"
+  - cmd: "%CMD_IN_ENV% python setup.py bdist_wininst"
+  - cmd: "%CMD_IN_ENV% python setup.py bdist_msi"
+  - ps: "ls dist"
+
+artifacts:
+  # Archive the generated packages in the ci.appveyor.com build report.
+  - path: dist\*
+
+matrix:
+  fast_finish: true
+
+notifications:
+  - provider: GitHubPullRequest
+    auth_token:
+      secure: iXYaAwXJcHdnRGTeb1Gk1PembhlZleCGDK2oJgsiFvA=
```

Comment:
Is it the right location for this token ? Should this be added in the settings associated with the project ?